internal/controller/resource_manager.go

```diff
@@ -147,7 +155,14 @@ func (r *ResourceManager) watchEvents(ctx context.Context, resource string, apiG
 			return err
 		}
 		metadataObj := convertMetaObjectToMetadata(*convertedData, resource)
-
+		if resource == "pods" {
```

Comment:
Same here, move that logic into `convertMetaObjectToMetadata`.
You shouldn't have to re-query data you already have just above (`resource`).